packages/@react-aria/overlays/src/usePopover.ts

```diff
@@ -92,8 +92,18 @@ export function usePopover(props: AriaPopoverProps, state: OverlayTriggerState):
     onClose: null
   });
 
+  // Delay preventing scroll until popover is positioned to avoid extra scroll padding.
+  // This requires a layout effect so that positioning has been committed to the DOM
+  // by the time usePreventScroll measures the element.
+  let [isPositioned, setPositioned] = useState(false);
+  useLayoutEffect(() => {
+    if (placement) {
+      setPositioned(true);
+    }
+  }, [isNonModal, placement]);
+
   usePreventScroll({
-    isDisabled: isNonModal
+    isDisabled: !isPositioned
```

Comment:
```suggestion
```
We'd like to test this PR with the useLayoutEffect removed and usePreventScroll() always set. Looking at related PRs we think the issues are related to iPad async example.